packages/client/src/query/aggregations/AggregationsClause.ts

```diff
@@ -38,7 +39,8 @@ export type AggregationClause<
     K
   >[P]["type"] extends "string"
     ? StringAggregateOption | StringAggregateOption[]
-    : ObjectTypePropertyDefinitionsFrom<O, K>[P]["type"] extends "double"
+    : ObjectTypePropertyDefinitionsFrom<O, K>[P]["type"] extends
+      "double" | "integer" | "float" | "decimal" | "byte" | "long" | "short"
       ? NumericAggregateOption | NumericAggregateOption[]
-    : ObjectTypePropertyDefinitionsFrom<O, K>[P]["type"];
+    : never;
```

Comment:
This old one was wrong because you cant have an aggregation of `{ geohash: "geopoint" }`. Switching to never to be more accurate.